.github/workflows/ur-build-hw.yml

```diff
@@ -0,0 +1,125 @@
+name: UR - Build adapters on HW
+
+on:
+  workflow_call:
+    inputs:
+      adapter_name:
+        required: true
+        type: string
+      other_adapter_name:
+        required: false
+        type: string
+        default: ""
+      runner_name:
+        required: true
+        type: string
+      platform:
+        description: "Platform string, `UR_CTS_ADAPTER_PLATFORM` will be set to this."
+        required: false
+        type: string
+        default: ""
+      static_loader:
+        required: false
+        type: string
+        default: OFF
+      static_adapter:
+        required: false
+        type: string
+        default: OFF
+
+permissions:
+  contents: read
+
+env:
+  UR_LOG_CUDA: "level:error;flush:error"
+  UR_LOG_HIP: "level:error;flush:error"
+  UR_LOG_LEVEL_ZERO: "level:error;flush:error"
+  UR_LOG_NATIVE_CPU: "level:error;flush:error"
+  UR_LOG_OPENCL: "level:error;flush:error"
+
+jobs:
+  adapter-build-hw:
+    name: Build & CTS
+    # run only on upstream; forks won't have the HW
+    if: github.repository == 'intel/llvm'
+    strategy:
+      matrix:
+        adapter: [
+          {
+            name: "${{inputs.adapter_name}}",
+            other_name: "${{inputs.other_adapter_name}}",
+            platform: "${{inputs.platform}}",
+            static_Loader: "${{inputs.static_loader}}",
+            static_adapter: "${{inputs.static_loader}}"
+          }
+        ]
+        build_type: [Release]
+        compiler: [{c: gcc, cxx: g++}]
+
+    runs-on: ${{inputs.runner_name}}
+
+    steps:
+    # TODO: If UR is merged into llvm it will require changes:
+    # - checkout llvm instead of UR
+    # - `cd sycl` and configure UR project from local tree
+    #
+    # Also, the step of downloading DPC++ should be integrated somehow;
+    # either compile it or use cached binaries (if accessible)...?
+    #
+    - name: Checkout UR
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        repository: 'oneapi-src/unified-runtime'
+        ref: main
```

Comment:
Once the move has been made the UR source will exits in the `unified-runtime` subdirectory of the intel/llvm checkout, can we specify the checkout to be placed there to validate the rest of the workflow?

Reply:
I updated the checkout place via `path: unified-runtime`, not sure if this is change you had in mind.

Reply:
Yes, this makes sense. Once we pull in the UR source this should be the same path so the rest of the workflow shouldn't need to change.